File: apps/go/routes/body/binary.go
package main

import "github.com/primate-run/go/route"

var _ = route.Post(func(request route.Request) any {
	data, mime, err := request.Body.Binary()
	if err != nil {
		return map[string]any{"error": err.Error()}
	}

	head := []int{}
	for i := 0; i < len(data) && i < 4; i++ {
		head = append(head, int(data[i]))
	}

	return map[string]any{
		"type": mime,
		"size": len(data),
		"head": head,
	}
})

File: apps/python/routes/body/binary.py
from primate import Route


@Route.post
def handle_post(request):
    bin_data = request.body.binary()

    return {
        "type": bin_data.content_type or request.headers.try_get("content-type"),
        "size": bin_data.size,
        "head": bin_data.head(4),
    }
